src/comm.cpp

```diff
@@ -157,8 +157,16 @@ void lbann_comm::allreduce(AbsDistMat& m,
 #ifdef LBANN_HAS_GPU
   if (m.GetLocalDevice() == El::Device::GPU) {
 #ifdef AL_HAS_NCCL
-    // Force GPU matrices to use NCCL.
+    // We require NCCL for GPU matrices.
```

Comment:
Aren't we able to handle the case where we have MPI-CUDA without NCCL, even if it's suboptimal?